message.go

```diff
@@ -362,13 +362,13 @@ type MessageProperties struct {
 
 	// The to field identifies the node that is the intended destination of the message.
 	// On any given transfer this might not be the node at the receiving end of the link.
-	To string
+	To *string
```

Comment:
While we're making breaking changes, I believe that some of these types are also incorrect per spec.  E.g. `To` is type `*` ([reference](https://github.com/Azure/go-amqp/blob/master/message.go#L336)) which means it can be [any type](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#doc-idp115568).  I think there might be a few others.

Reply:
This one is interesting. So if we dig in it's less clear to me what the type should be:

This field does appear to have type '*', but that 'requires' field applies a restriction to the type:
```xml
<field name="to" type="*" requires="address"/>`
```

Address is defined like this:
```xml
<type name="address-string" class="restricted" source="string" provides="address"/>
```

Which, I believe, means that:
1. 'address' is really a type 'address-string'
2. 'address-string' is derived from 'string'

So here's where it gets a bit more interesting:

- qpid (a Java AMQP stack) defines it as type string: [qpid](https://qpid.apache.org/releases/qpid-proton-j-0.33.10/api/org/apache/qpid/proton/amqp/messaging/Properties.html) 
- Azure AMQP defines it as an `Address` type [azure-amqp](https://github.com/Azure/azure-amqp/blob/4b347c370823d5d0bd95c193f3e018501ef3847f/src/Framing/Address.cs#L38) but it looks like they basically only handle 'string' as well through their AddressString type (same file, it's a concrete implementation of Address).

So it seems right to define it as a string. For that one type. And then we have to go through the other types to see if they boil down the same way.

Reply:
Hmm I see.  We can hold off on making any changes to the field types for now since it's not as straight-forward as I thought it might be.

Reply:
I think it's better than we think, now that I look at it more. 

So message-id, for instance is defined like this: (I added in the 'type' translation, but it matches what we have):

```go
// - uint64:    <type name="message-id-ulong" class="restricted" source="ulong" provides="message-id"/>
// - amqp.UUID: <type name="message-id-uuid" class="restricted" source="uuid" provides="message-id"/>
// - []byte:    <type name="message-id-binary" class="restricted" source="binary" provides="message-id"/>
// - string:    <type name="message-id-string" class="restricted" source="string" provides="message-id"/>
```

So the 'message-id' types domain includes 4 different primitives, basically.

So I think we're actually correct with address being a string type since only one 'concrete' type exists for 'address' ('address-string'). If we say that AMQP 1.0 is always what our public facing surface is then we're in good shape. That's probably the right call otherwise _everything_ will have to be interface{}, which would be worse, I think.